### ISSUE TITLE
what about a `mkdir` flag?

### ISSUE THREAD

**Issue by jakdept** [sebdah/goldie#3]
Love this.

One minor annoyance - when I add a completely new test, I have to create a directory within my fixtures directory by hand ex:

In this example, `FixtureDir` is `testdata/fixtures`, and in this case I just added a table test called `TestThumbnailJPG` with subtests `TestThumbnail-jpg-0` etc. So, then when i `go test -update` it still fails:

```
--- FAIL: TestThumbnailJPG (2.88s)
    --- FAIL: TestThumbnailJPG/TestThumbnail-jpg-0- (0.10s)
    	goldie.go:56: open testdata/fixtures/TestThumbnailJPG/TestThumbnail-jpg-0-.golden: no such file or directory
    --- FAIL: TestThumbnailJPG/TestThumbnail-jpg-1- (0.08s)
    	goldie.go:56: open testdata/fixtures/TestThumbnailJPG/TestThumbnail-jpg-1-.golden: no such file or directory
    --- FAIL: TestThumbnailJPG/TestThumbnail-jpg-2- (0.09s)
    	goldie.go:56: open testdata/fixtures/TestThumbnailJPG/TestThumbnail-jpg-2-.golden: no such file or directory
```

The `-update` flag fails because the directory for that test does not exist. So, I `mkdir` then it goes well.

It'd be nice to have a `-mkdir` flag that assumes `-update`, or maybe `-update mkdir` as an option command to the `-update` flag or something - so when you run that, it automatically creates the required folders. I'd only see a use for that flag when you've changed `FixtureDir` or when you've written a new test - with a new test name - so I think I'd rather it not happen every time you `-update`.

**Comment by sebdah** [comment]
Oh, that's a good one. Maybe simply always create any non-existing folders if the fixture contains a `/` (or another sign depending on OS).

**Comment by jakdept** [comment]
I don't know about all of the time - maybe that makes more sense, but there's times that I have to `-update` because someone upstream updated a library, which changes my results - in that case I wouldn't want to make directories.

On the other hand, I think that's rare enough that it's not such a big deal? For sure though, not having to create directories would be nice.

**Comment by jakdept** [comment]
Yeah, even better you could split off the dir with `filepath.Dir(` and then stat like you're doing in `ensureFixtureDir(` and maybe `os.MkdirAll(` instead of `os.Mkdir(`.

If you wanted to add a flag for folder creation, that's just an extra bit. I can give you PRs with a flag or without, if you'd like.

**Comment by sebdah** [comment]
Sorry, all the time was a little to broad. I ment "whenever the folder doesn't exist". I don't see the harm in that. My preference would be to do this without a flag to optimize for ease of use. If it turns out to become an issue we could always expand with a flag later on.

**Comment by sebdah** [comment]
And also, if you'd like to contribute with a PR that'd be fantastic! I can review and merge it whenever I get it.

**Comment by jakdept** [comment]
Resolved in PR #4 🎉
